modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportCGC.pm

```diff
@@ -76,60 +78,45 @@ sub fetch_input {
-  my $cgc_google_url = 'https://storage.googleapis.com/open-targets-data-releases/';
-  # example of URL format: https://storage.googleapis.com/open-targets-data-releases/18.12/output/18.12_evidence_data.json.gz';
-
   open(my $logFH, ">", $workdir."/".'log_import_out_'.$source_info{source_name_short}.'_'.$species) || die ("Failed to open file: $!\n");
   open(my $errFH, ">", $workdir."/".'log_import_err_'.$source_info{source_name_short}.'_'.$species) || die ("Failed to open file: $!\n");
   open(my $pipelogFH, ">", $workdir."/".'log_import_debug_pipe_'.$source_info{source_name_short}.'_'.$species) || die ("Failed to open file: $!\n");
   $self->logFH($logFH);
   $self->errFH($errFH);
   $self->pipelogFH($pipelogFH);
 
-  #get input file CGC via OpenTargets, get latest published file:
+  # get current date
   my $dt = DateTime->now;
   my $year = sprintf("%02d", $dt->year % 100);
   my $month = sprintf ("%02d",$dt->month);
-  my $cgc_ftp_url = $cgc_google_url.$year.'.'.$month.'/output/'.$year.'.'.$month.'_evidence_data.json.gz';
 
-  my $file_opent = basename($cgc_ftp_url); #OpenTargets file
-  print $logFH "INFO: Found file ($file_opent), will skip new fetch\n" if -e $workdir."/".$file_opent;
-  my $found = (-e $workdir."/".$file_opent ? 1: 0);
+  # get input file CGC via OpenTargets, get latest published file:
+  my $script_dir = $repo_dir . "/ensembl-variation/scripts/python";
+  my $file_opent = "cgc_input_latest.json";
+  system("python3 $script_dir/download_cgc_file.py -d $workdir -r latest") unless -e $workdir."/".$file_opent;
 
-  while (!$found){
-    if ($month == 01){ $month =12; $year = $year-1;}
-    else { $month = $month -1;}
-    $month = sprintf ("%02d",$month);
-    my $cgc_check_url = $cgc_google_url.$year.'.'.$month.'/output/'.$year.'.'.$month.'_evidence_data.json.gz';
-    $file_opent = basename($cgc_check_url);
+  my $found = (-e $workdir."/".$file_opent ? 1: 0);
 
-    if (-e $workdir."/".$file_opent){
-      $found = 1;
-      print $logFH "INFO: Found $file_opent, will skip new fetch.\n";
-      next;
-    }
+  if(!$found) {
+    print $errFH "Input file ($file_opent) is missing\n";
+    die "Input file is missing: check download script download_cgc_file.py\n";
+  }
 
-    my $resHTTPcode = qx{curl -w %{http_code} -X GET $cgc_check_url -o '$workdir/$file_opent'};
-    if ($resHTTPcode == 404){ #Not Found
-      unlink($workdir."/".$file_opent) if -e $workdir."/".$file_opent;
-      print $errFH "WARNING: No OpenTarget file found for current month ($file_opent), will check past month.\n";
-    } elsif ($resHTTPcode == 200){ #Successful
-      print $errFH "HTTP code 200 (successful) but file ($file_opent) is missing \n" unless -e $workdir."/".$file_opent;
-    }
-    if (-e $workdir."/".$file_opent) { $found = 1;}
+  print $logFH "INFO: Found input file ($file_opent)\n" ;
+  my $valid_file = $self->validate_input_file($file_opent);
+  if(!$valid_file) {
+    print $errFH "Input file ($file_opent) has incorrect format\n";
```

Comment:
Suggesting error message should be 
"Input file ($file_opent) has incorrect format or empty \n";